index.js

```diff
@@ -57,6 +57,45 @@ module.exports = {
     // called and that is dynamic, so we need to make sure we are requiring the correct parent file everytime.
     delete require.cache[__filename];
 
+    // Finally return the target file's config.
+    return require(targetFile);
+  },
+
+  /**
+   * `delete require.cache()` might not work properly if there is any hook applied to require module, eg: require-in-the-middle
+   * switchV2() will expect parentFilename as an parameter and use that to compare with the targetFile.
+   *
+   * It will pick the env config based on the SAILS_ENV value. If a project has SAILS_ENV
+   * defined to some value then it expects that same file in the env/config file so that it
+   * will pick the config value from that file.
+   *
+   * @param env this is the current sails environment in which this module is used.
+   * @param parentFilename this is the filename of the module which is calling switchV2.
+   * @param defaultConfig this is the default config which is bases on NODE_ENV value
+   * @returns {config}
+   *
+   * @example
+   * // This is the env config file
+   * module.exports = require('sails-env-switch').switchV2(__filename, {
+   *    port: 8080,
+   *    name: 'Sample'
+   * });
+   */
+  switchV2: function (parentFilename, defaultConfig) {
```

Comment:
Imo the stack approach is better, this we can keep the interface same. 
Why didn't we go with that ?

If need be switch and switchV2 can be merged. 
https://stackoverflow.com/a/63039252

cc: @kunagpal